content/oc/0001-fundamentos-computacionais.md

```diff
@@ -0,0 +1,194 @@
+$$
+
+$$
+CPUTime = \frac{CPUClockCycles}{ClockRate}
+$$
+
+Resumidamente, o que estamos a calcular é o **tempo que demoramos através de quantos ciclos demos e quanto tempo demoramos por ciclo** ou o seu inverso, ou seja quantos ciclos demos sobre a frequência de cada ciclo. Assim, podemos facilmente concluir que a nossa _performance_ poderá ser melhorada através da [redução do número de ciclo de relógios](color:pink), simplificando o algoritmo ou tendo um processador mais inteligente; [aumentando o _clock rate_](color:pink), a frequência com que funciona ou até mesmo com um [_hardware_ mais adequado](color:pink) que troca o _clock rate_ com o _cycle count_.
+
+:::info[Exemplo]
+
+Imaginando que temos um Computador A com 2GHz de relógio, 10 segundo de tempo de CPU e queremos desenhar um Computador B com [6 segundos](color:orange) de tempo de CPU e tem um [relógio mais rápido](color:orange) mas faz com que os [ciclos de relógio sejam 1.2 vezes maiores](color:orange), o **quão rápido é que conseguimos ue o nosso relógio de B seja?**
+
+Começando por calcular o _clock rate_ de B, temos:
+
+$$
+ClockRate_B = \frac{ClockCycles_B}{CPUTime_B}=\frac{1,2*ClockCycles_A}{6s}
+$$
+
+$$
+ClockCycles_A = CPUTime_A * ClockRate_A = 10s * 2GHz = 20 * 10⁹
+$$
+
+$$
+Sup = \frac{10s}{6s} = \frac{t_A = numCycle_A * T_{CB}}{t_B = numCycle_B * T_{CB}} = \frac{numCycle * T_{CA}}{1,2 * numCycle * T_{CB}} = \frac{6/10}{0,6}
+$$
+
+$$
+ClockRate_B = \frac{1,2 * 20 * 10⁹}{6s} = \frac{24*10⁹}{6s} = 4GHz
+$$
+
+:::
+
+### Número de instruções e ciclos por instrução
+
+O nosso [número de instruções](color:pink) é determinado pelo programa, ISA e o compilador, enquanto o [CPI](color:pink) é determinado pelo CPU _hardware_. Este último, caso tenha diferentes instruções, tem que ter diferentes CPIs, o CPI médio é afetado por uma mistura de instruções.
+
+Ficamos, portanto com as seguintes fórmulas:
+
+$$
+ClockCycles = InstructionCount * CyclesPerInstruction
+$$
+
+$$
+CPUTime = InstructionCount * CPI * ClockCycleTime = \frac{InstructionCount * CPI}{ClockRate}
+$$
+
+Então, se tivermos diferentes classes de instruções, temos um número diferente de ciclos:
+
+$$
+ClockCycles = \sum^n_{i=1} (CPI_i * InstructionCount_i)
+$$
+
+E, para a média ponderada do CPI:
+
+$$
+CPI = \frac{ClockCycles}{InstructionCount} = \sum^n_{i=1} (CPI_i * \frac{InstructionCount_i}{InstructionCount}
+$$
+
+Vendo um exemplo, se dissermos que uma determinada quantidade de alunos já não tem que pagar as suas proprinas, para calcularmos a média ponderada que irá ficar paar cada aluno, então temos que ver cada aluno, quanto ele paga e a sua relação com o total de todos os alunos.
+
+## MIPS
+
+O [MIPS](color:pink), _Millions of Instructions per Second_ é uma arquitetura de microprocessadores RISC usada na maior parte dos processadores hoje em dia. Contudo, esta arquitetura não tem em conta as diferenças entre os ISA dos computadores nem a complexidade entre as instruções.
```

Comment:
The *MIPS* architecture is referred to as _Millions of Instructions per Second_. However, it stands for _Microprocessor without Interlocked Pipelined Stages_ as stated in this [Wikipedia page](https://en.wikipedia.org/wiki/MIPS_architecture). _Millions of Instructions per Second_ is not an architecture, but rather a performance metric.
```suggestion
O [MIPS](color:pink), _Microprocessor without Interlocked Pipelined Stages_ é uma arquitetura de microprocessadores RISC usada na maior parte dos processadores hoje em dia. Contudo, esta arquitetura não tem em conta as diferenças entre os ISA dos computadores nem a complexidade entre as instruções.
```

Reply:
It appears that she got the name right, but the explanation is wrong (it should be about the performance metric and not the CPU architecture)